### PR TITLE
Submitter supplied protocol from the API

### LINKED ISSUE #225: Submitter-processed protocol information appears to be hardcoded
All submitter-processed protocol information appears to be the same, regardless of the experiment in question.

`GSM182283` from `GSE7529` 

screen shot 2018-08-07 at 2 35 31 pm

`GSM399040` from `E-GEOD-15892`

screen shot 2018-08-07 at 2 37 47 pm

I believe this may be hardcoded?

https://github.com/AlexsLemonade/refinebio-frontend/blob/c15940459fdd39072e5c05089129f4d06990550d/src/containers/Experiment/SamplesTable.js#L349

We'll need to supply the protocol information for the sample that the user has clicked on. The information on [this page](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM399040) is what I would expect to see for `GSM399040`.

### LINKED ISSUE #319: Processing information column of samples table includes SalmonTools for RNA-seq samples
We only want to display `Salmon quant` and `tximport`.

sample_table_for_rna-seq

### PR DESCRIPTION
**Pull request by arielsvn**
## Issue Number

close #225 
close #319 

## Purpose/Implementation Notes

Refactored the submitter processed protocol info and shows the correct fields from `processed_info`, depending on the type of sample.

For now the processed information is displayed only for **GEO** and **SRA** samples.

## Types of changes

* [x] New feature (non-breaking change which adds functionality)
* [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Functional tests

List out the functional tests you've completed to verify your changes work locally.

## Checklist

* [ ] Lint and unit tests pass locally with my changes
* [ ] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)
* [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots

image

image

### CODE REVIEW

**Comment by arielsvn** [comment]
Updating the title, this PR should be ready to be reviewed. I also addressed https://github.com/AlexsLemonade/refinebio-frontend/issues/319 here.

Once it get's merged we'll be able to re-use the same components for https://github.com/AlexsLemonade/refinebio-frontend/issues/293

Geo Samples:

image

SRA:

image

Array Express:

File: src/components/Accordion.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import classnames from 'classnames';
+import Button from './Button';
+import Checkbox from './Checkbox';
+import './Accordion.scss';
+
+export class Accordion extends React.Component {
+  state = {
```
[the remark this answers is not in the document]
Comment (arielsvn) [reply src/components/Accordion.js]: This is equivalent to initializing `state` inside the constructor. It's a syntax feature of EcmaScript, supported by the "Class Fields and Static Properties" proposal. Which is enabled by default by [create-react-app](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#supported-language-features-and-polyfills)

File: src/components/Accordion.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    activeElements: []
+  };
+
+  componentDidMount() {
+    this.setState({
+      activeElements: new Array(this._totalChildren()).fill(false)
```
[the remark this answers is not in the document]
Comment (arielsvn) [reply src/components/Accordion.js]: It is the initial value, to calculate it we need to be able to access the element's `children` property. I had the impression that it wasn't good to use this property in the constructor, but I can't find any references now. Let me give it a try.

**Comment by arielsvn** [comment]
Thanks for the review @dongbohu! I updated the initialization of `state`, and placed it inside the constructor as you suggested. It's way better now!

File: src/components/Accordion.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dongbohu) [reply src/components/Accordion.js]: Wow, I didn't know that. It is not easy to keep track of new features in ECMAScript.

**Pull request merged**